src/libraries/System.Net.Http.Json/tests/FunctionalTests/HttpContentJsonExtensionsTests.cs

```diff
@@ -0,0 +1,287 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Test.Common;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Http.Json.Functional.Tests
+{
+    public class HttpContentJsonExtensionsTests
+    {
+        private readonly List<HttpHeaderData> _headers = new List<HttpHeaderData> { new HttpHeaderData("Content-Type", "application/json") };
+
+        [Fact]
+        public async Task ThrowOnNull()
+        {
+            HttpContent content = null;
+            await Assert.ThrowsAsync<ArgumentNullException>(() => content.ReadFromJsonAsync<Person>());
+            await Assert.ThrowsAsync<ArgumentNullException>(() => content.ReadFromJsonAsync(typeof(Person)));
+        }
+
+        [Fact]
+        public async Task HttpContentGetThenReadFromJsonAsync()
+        {
+            const int NumRequests = 2;
+            await LoopbackServer.CreateClientAndServerAsync(
+                async uri =>
+                {
+                    using (HttpClient client = new HttpClient())
+                    {
+                        var request = new HttpRequestMessage(HttpMethod.Get, uri);
+                        var response = await client.SendAsync(request);
```

Comment:
Please replace the use of var anywhere the RHS isn't a `new` or an explicit cast.

Also, as noted earlier, we should be Dispose'ing of all HttpResponseMessages.